tests/acceptance/30_custom_promise_types/05_meta_attr.cf

```diff
@@ -0,0 +1,99 @@
+######################################################
+#
+#  Basic test of custom promise types / promise modules checking that using a
+#  'meta' attribute in combination with a custom promise works (i.e. doesn't
+#  crash).
+#
+#####################################################
+body common control
+{
+    inputs => { "../default.cf.sub" };
+    bundlesequence  => { default("$(this.promise_filename)") };
+    version => "1.0";
+}
+
+#######################################################
+
+bundle agent init
+{
+  files:
+    "$(G.testfile)"
+      delete => init_delete;
+}
+
+body delete init_delete
+{
+      dirlinks => "delete";
+      rmdirs   => "true";
+}
+
+#######################################################
+
+promise agent example
+{
+    interpreter => "/bin/bash";
+    path => "$(this.promise_dirname)/example_module.sh";
+}
+
+body classes example
+{
+        promise_repaired => { "example_promise_repaired" };
+}
+
+bundle agent test
+{
+  meta:
+    "description" -> { "CFE-3440" }
+      string => "Test that you can use a meta attribute with a custom promise";
+
+  vars:
+    "test_string"
+      string => "hello, modules";
+
+  example:
+    cfengine::
```

Comment:
Why use this class guard?

Reply:
Because I copied a test that had it. I guess it was added there to make sure class guards in general don't break custom promises. Doesn't hurt.